src/components/tools/Partials/ChangeLocation.svelte

```diff
@@ -35,9 +35,15 @@
   let showSuggestions = false;
 
   async function mapClick(e) {
-    const addresses = await reverseGeocode(`${e.detail[0]}, ${e.detail[1]}`);
-    const feature = addresses.features[0];
-    currentLoc = await getFeature(feature, currentBoundary.id);
+    let newLocation;
```

Comment:
@mukhtyar from what I can tell the call to `reverseGeocode` here isn't needed, since ultimately a lng, lat coordinate is being passed to `getBoundaryPolygon` (in `/helpers/geocode/` to find the appropriate geometry for the selected boundary type. Let me know if there's a case I'm missing.

Reply:
Edit: the one place where this changes things is that when a user selects a "place" boundary type when they click outside of a place polygon nothing will happen, which seems fine IMO.

Reply:
The place name returned by `reverseGeocode`  is used as the default location title if the user is using LOCA boundary. See functions `formatFeature` and `getTitle`. If user has selected a boundary other than LOCA, the location title is derived from the boundary polygon properties returned by API. But the `reverseGeocode` lookup could be moved so it only happens for LOCA boundaries.

Reply:
I'm thinking that perhaps getting a place name for the title should be separate from selecting the grid cell. I'm noticing an odd behavior where sometimes clicking on a grid cell doesn't actually select it, but will select a neighboring grid cell. I believe this is because for certain cases the lat lon gets reverse geocoded to a neighboring place and thus neighboring grid cell when the grid cell being clicked on doesn't have a place associated with it in the Mapbox geocoding API.

Perhaps it makes sense to pass the coordinates from the map click directly to `getFeature` and then do the reverse geocoding once the user has confirmed their selection?

Reply:
That makes sense to me.